ODIN_II/SRC/partial_map.cpp

```diff
@@ -1134,14 +1134,32 @@ void instantiate_arithmetic_shift_right(nnode_t *node, short mark, netlist_t *ne
 	}
 	buf_node = make_1port_gate(BUF_NODE, width, width, node, mark);
 	/* connect inputs to outputs */
-	for(i = width - 2; i >= shift_size; i--)
+	for(i = width - 1; i >= shift_size; i--)
 	{
 		// connect higher input pin to lower output pin
 		remap_pin_to_new_node(node->input_pins[i], buf_node, i-shift_size);
 	}
-	/* connect first pin to outputs that don't have inputs connected */
-	i = width - 1;
-	remap_pin_to_new_node_range(node->input_pins[i], buf_node, i, i-shift_size);
+	
+	int pad_bit = width - 1 - shift_size;
+
+	/* connect ZERO to outputs that don't have inputs connected */
```

Comment:
change the comment to reflect the fact that you use the pad pin

Reply:
Not sure if new comment is clear enough?